MoreLinq.Test/TestingSequence.cs

```diff
@@ -21,26 +21,38 @@ namespace MoreLinq.Test
     using System.Collections;
     using System.Collections.Generic;
     using NUnit.Framework;
+    using static TestingSequence;
 
     static class TestingSequence
     {
         internal static TestingSequence<T> Of<T>(params T[] elements) =>
-            Of(Options.None, elements);
+            new(elements, Options.None, numEnumerations: 1);
 
         internal static TestingSequence<T> Of<T>(Options options, params T[] elements) =>
-            elements.AsTestingSequence(options);
+            elements.AsTestingSequence(options, numEnumerations: 1);
 
         internal static TestingSequence<T> AsTestingSequence<T>(this IEnumerable<T> source,
-                                                                Options options = Options.None) =>
+                                                                Options options = Options.None,
+                                                                int numEnumerations = 1) =>
             source != null
-            ? new TestingSequence<T>(source) { IsReiterationAllowed = options.HasFlag(Options.AllowMultipleEnumerations) }
+            ? new TestingSequence<T>(source, options, numEnumerations)
             : throw new ArgumentNullException(nameof(source));
 
+        internal const string ExpectedDisposal = "Expected sequence to be disposed.";
+        internal const string TooManyEnumerations = "Sequence should not be enumerated more than expected.";
+        internal const string TooManyDisposals = "Sequence should not be disposed more than once per enumeration.";
+        internal const string SimultaneousEnumerations = "Sequence should not have simultaneous enumeration.";
+        internal const string MoveNextDisposed = "LINQ operators should not call MoveNext() on a disposed sequence.";
```

Comment:
nit: use `nameof`:

```suggestion
        internal const string MoveNextDisposed = $"LINQ operators should not call {nameof(IEnumerator<object>.MoveNext)} on a disposed sequence.";
```

Reply:
I'm all for using `nameof` in most cases, but in this case, I feel like it reduces readability (by adding references and symbols to get to `MoveNext`). 

`nameof` was introduced to add security for renaming of class/struct/interface members, but `MoveNext` is part of the framework and will not be changed (per breaking change requirements of dotnet/runtime), so that security is not needed here. 

Leaving as is for now, unless further requested.